includes/classes/class-rest-api.php

```
@@ -643,4 +653,48 @@ public function get_site_summary( \WP_REST_Request $request ) {
+	 * @param string           $capability The capability to check.
+	 * @return \WP_Error|boolean
+	 */
+	public static function check_token_or_nonce_and_capability_permissions_check( $request, $capability = 'manage_options' ) {
+		$token = sanitize_text_field( $request->get_header( 'X-EDAD-Token' ) );
+		// If we have a token then it takes priority, check it, otherwise fallback to nonce.
+		if ( method_exists( 'EDAC\Inc\REST_Api', 'api_token_verify' ) && $token ) {
+			if ( ! self::api_token_verify( $token ) ) {
+				return new \WP_Error( 'rest_forbidden', __( 'Invalid token.', 'accessibility-checker' ), [ 'status' => 401 ] );
+			}
+		} elseif ( $request->get_header( 'X-WP-Nonce' ) ?? $request->get_param( 'nonce' ) ) {
+			$nonce = sanitize_text_field( $request->get_header( 'X-WP-Nonce' ) ?? saitize_text_field( $request->get_param( 'nonce' ) ) );
```

PHPStan (2.0.3)

Function saitize_text_field not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(function.notFound)